### PR TITLE
Calculator component test

### PR DESCRIPTION
**Pull request by sagieramos**
### Unit testing for the following file using jest:
- `operate.js` and `calculate.js`.
- Used Jest snapshots to test the `Calculator` component.
- Used React Testing Library to simulate user interaction.
- All tests pass.

To execute test:
```sh
  npm test
```

### CODE REVIEW

**Review by V-Blaze — CHANGES_REQUESTED** [review]
Hi @sagieramos 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

dcentl8-9b96e129-9725-401b-ac34-2feb990202a0

## Invalid Code Review Request

## You have submitted a project, that does not have the commits and contribution of your partner. This is a pair programming activity. Both students need to contribute meaningfully to this branch.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/tests/Calculator.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  render(<Calculator />);
+  const calculator = screen.getByTestId('calc-wrapper');
+  expect(calculator).toMatchSnapshot();
+});
+test('should update the calculator input when the user presses a number button', () => {
+  render(<Calculator />);
+  const input = screen.getByRole('textbox');
+  fireEvent.click(screen.getByText('1'));
+  expect(input.value).toBe('1');
+});
```
Comment (danmainah) [inline src/tests/Calculator.test.js:21]: - [x] @Team Great work adding tests for the Calculator component, however you have not added tests for all the other components which is part of the project requirements
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @sagieramos and @Uzair-Manzoor , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Added tests for the operate.js and calculate.js file using jest.
✔️  Ensured all tests added are passing.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Kindly consider adding how to run tests in your getting stated section in the readme file.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Efiamotu-1 — APPROVED** [review]
Hi @Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# STATUS: APPROVED 🕺🥳 

image

## Highlights :dart: 
* Correct GitHub flow or Gitflow :heavy_check_mark: 
* Good commit messages :heavy_check_mark: 
* All linters passing :heavy_check_mark: 
* Good PR summary :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, you can now Merge since it is Approved ✅, However if there are any conflicts please resolve them before merging.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**